stmhal/dma.c

```diff
@@ -80,7 +80,7 @@ static int get_dma_id(DMA_Stream_TypeDef *dma_stream) {
     }
 }
 
-void dma_init(DMA_HandleTypeDef *dma, DMA_Stream_TypeDef *dma_stream, uint32_t dma_channel, uint32_t direction, void *data) {
+void dma_init(DMA_HandleTypeDef *dma, DMA_Stream_TypeDef *dma_stream, DMA_InitTypeDef dma_init, uint32_t dma_channel, uint32_t direction, void *data) {
```

Comment:
It should take a pointer to the DMA_InitTypeDef to avoid copying a large amount of data for the function call.